experiment/manual-trigger/manual-trigger.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// manual-trigger triggers jenkins jobs based a specified github pull request
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"strings"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/jenkins"
+	"k8s.io/test-infra/prow/kube"
+)
+
+type options struct {
+	githubEndpoint     string
+	githubToken        string
+	jenkinsBearerToken string
+	jenkinsEndpoint    string
+	jobName            string
+	num                int
+	org                string
+	repo               string
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.githubEndpoint, "githubapiurl", "https://api.github.com", "GitHub's API endpoint.")
+	flag.StringVar(&o.jenkinsEndpoint, "jenkinsurl", "", "Jenkins url.")
+
+	flag.StringVar(&o.githubToken, "ghtoken", "", "Path to file containing GitHub token.")
+	flag.StringVar(&o.jenkinsBearerToken, "jbtoken", "", "Path to file containing Jenkins Bearer token.")
```

Comment:
[`jenkins-bearer-token-file`](https://github.com/kubernetes/test-infra/blob/4dd7e4dcd350e767e1418b2b815a5914e7943daa/prow/cmd/jenkins-operator/main.go#L43)

Do we want to support basic auth?